File: sample-app/tests/integration_test.py
#!/usr/bin/env python3
import requests
import sys

def test_health_endpoint(base_url):
    """Test the health endpoint returns 200 OK"""
    try:
        response = requests.get(f"{base_url}/health", timeout=5)
        assert response.status_code == 200, f"Expected 200, got {response.status_code}"
        assert response.text.strip() == "OK", f"Expected 'OK', got '{response.text.strip()}'"
        return True
    except Exception as e:
        print(f"Health check failed: {e}")
        return False

def test_root_endpoint(base_url):
    """Test the root endpoint returns expected message"""
    try:
        response = requests.get(base_url, timeout=5)
        assert response.status_code == 200, f"Expected 200, got {response.status_code}"
        assert "Hello World" in response.text, f"Expected 'Hello World' in response"
        return True
    except Exception as e:
        print(f"Root endpoint test failed: {e}")
        return False

if __name__ == "__main__":
    base_url = sys.argv[1] if len(sys.argv) > 1 else "http://localhost:8080"
    
    tests = [
        ("Health Endpoint", test_health_endpoint),
        ("Root Endpoint", test_root_endpoint)
    ]
    
    all_passed = True
    for test_name, test_func in tests:
        print(f"Running {test_name}...", end=" ")
        if test_func(base_url):
            print("✓ PASSED")
        else:
            print("✗ FAILED")
            all_passed = False
    
    sys.exit(0 if all_passed else 1)

File: sample-app/integration_test.go
package main

import (
	"io"
	"net/http"
	"os"
	"strings"
	"testing"
	"time"
)

var baseURL string

func init() {
	baseURL = os.Getenv("BASE_URL")
	if baseURL == "" {
		baseURL = "http://localhost:8080"
	}
}

func TestHealthEndpoint(t *testing.T) {
	client := &http.Client{Timeout: 5 * time.Second}

	resp, err := client.Get(baseURL + "/health")
	if err != nil {
		t.Fatalf("Health check failed: %v", err)
	}
	defer resp.Body.Close()

	if resp.StatusCode != 200 {
		t.Errorf("Expected status 200, got %d", resp.StatusCode)
	}

	body, err := io.ReadAll(resp.Body)
	if err != nil {
		t.Fatalf("Failed to read response body: %v", err)
	}

	if strings.TrimSpace(string(body)) != "OK" {
		t.Errorf("Expected 'OK', got '%s'", strings.TrimSpace(string(body)))
	}
}

func TestRootEndpoint(t *testing.T) {
	client := &http.Client{Timeout: 5 * time.Second}

	resp, err := client.Get(baseURL)
	if err != nil {
		t.Fatalf("Root endpoint test failed: %v", err)
	}
	defer resp.Body.Close()

	if resp.StatusCode != 200 {
		t.Errorf("Expected status 200, got %d", resp.StatusCode)
	}

	body, err := io.ReadAll(resp.Body)
	if err != nil {
		t.Fatalf("Failed to read response body: %v", err)
	}

	if !strings.Contains(string(body), "Hello World") {
		t.Errorf("Expected 'Hello World' in response, got: %s", string(body))
	}
}
